app/sdk/utils/branding/BrandingManager.java

```diff
@@ -0,0 +1,130 @@
+package sdk.utils.branding;
+
+import akka.stream.Materializer;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.net.util.Base64;
+import org.apache.http.client.utils.URIBuilder;
+import org.asynchttpclient.AsyncHttpClientConfig;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+import play.Application;
+import play.api.libs.ws.WSClientConfig;
+import play.api.libs.ws.ahc.AhcConfigBuilder;
+import play.api.libs.ws.ahc.AhcWSClientConfig;
+import play.api.libs.ws.ahc.AhcWSClientConfigFactory;
+import play.api.libs.ws.ssl.SSLConfigFactory;
+import play.libs.ws.WSClient;
+import play.libs.ws.WSRequest;
+import play.mvc.Http;
+import scala.concurrent.duration.Duration;
+import sdk.utils.JsonUtils;
+
+import javax.inject.Inject;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Author: Karis Sponsler
+ * Date: 11/22/17
+ */
+public class BrandingManager {
+    private static WSClient wsClient;
+    @Inject
+    private static Materializer materializer;
+
+    public static CompletionStage<Branding> getBranding(String appId, String version) {
+        return getBranding(appId, version, "https://go-monolith-v2-sbx.herokuapp.com");
+    }
+
+    public static CompletionStage<Branding> getBranding(String appId, String version, String routerUrl) {
+        WSRequest request;
+        try {
+            request = proxylessClient().url(
+                new URIBuilder(routerUrl)
+                    .setPath("/client/1/config/brandingConfig")
+                    .build()
+                    .toString()).setHeader("X-APPTREE-APPLICATION-ID", appId).setHeader("X-APPTREE-VERSION", version);
+        } catch (URISyntaxException e) {
+            return CompletableFuture.supplyAsync(() -> null);
+        }
+        return request
+            .get()
+            .thenApply(wsResponse -> {
+                // Use default branding (baseline page styles) if request failed
+                if (wsResponse.getStatus() != Http.Status.OK) {
+                    return null;
+                }
+
+                JsonNode json = wsResponse.asJson();
+                return Optional.ofNullable(JsonUtils.fromJson(json, Branding.class)).orElse(null);
```

Comment:
Doesn't `Optional.ofNullable` return value if not null or `Optional.empty()` if it is null?